docs/guides/checkpointing_solutions/Emergency_Checkpointing.md

```diff
@@ -0,0 +1,181 @@
+
+* `--enable-mtc`: Enables the Multi-Tier Checkpointing feature.
+* `--enable-gcsfuse-csi-driver`: Installs the required GCS FUSE CSI driver.
+* `--mtc-ramdisk-size`: Allocates an in-memory ramdisk on each node for fast, local checkpoints.
+* `--mtc-gcs-bucket`: Specifies the GCS bucket. It is not utilized in emergency checkpointing, but is needed to deploy checkpointing configurations.
+
+
+### Calculating Ramdisk Size Per Host 
+
+The total size of a full training checkpoint (including model weights and optimizer state) can be estimated based on the number of model parameters.
+A good rule of thumb for a model trained with:
+**Total Checkpoint Size ≈ Number of Parameters × 12 bytes**
+
+For example, a 1 billion parameter model would require approximately **1B × 12 bytes = 12 GB** for a full checkpoint.
+
+In a distributed training environment, the checkpoint is **sharded**, or split, across all the hosts in a slice. Each host is only responsible for saving its portion of the total checkpoint. Therefore, the ramdisk on a single pod only needs to be large enough for its local shard.
+
+The formula is:
+**Required Ramdisk Size per Pod ≈ Total Checkpoint Size / Number of Hosts in the Slice**
+
+It's a good practice to add a **10-15% buffer** .
+
+### Example Calculation
+
+Let's walk through an example for a large model.
+
+* **Model**: A 70 billion parameter language model.
+* **Training Slice**: A nodepool with **32 hosts**.
+
+1.  **Estimate Total Checkpoint Size**:
+    `70,000,000,000 parameters × 12 bytes/parameter = 840,000,000,000 bytes`
+    `840,000,000,000 bytes ≈ 840 GB`
+2.  **Calculate Per-Host Checkpoint shard**:
+    `(Total Checkpoint Size / 32 hosts) = 26.25 GB per host`
+
+3.  **Calculate Per-Host Ramdisk Size**:
+    `(Per-Host Checkpoint shard) * 2 = 52.50 GB per host`
+
+4.  **Add a Safety Buffer (e.g., 15%)**:
+    `(Per-Host Ramdisk Size) × 1.15 ≈ 60.3 GB`
+
+In this scenario, you should configure each pod in that slice with a ramdisk of at least **60 GB**.
+
+### Example XPK cluster creation command
+
+```
+OUTPUT_PATH=<gcs bucket output path>
+PROJECT_ID=<project id>
+ZONE=<your zone>
+CLUSTER_NAME=<cluster name>
+TPU_TYPE=<tpu-type>
+MACHINE_TYPE=<cpu machine-type>
+NUM_SLICES=<number of slices>
+RAMDISK_SIZE="60000Mi"
+GKE_VERSION=1.32.3-gke.1785000
+
+gcloud config set project ${PROJECT_ID}
+gcloud config set compute/zone ${ZONE}
+
+git clone https://github.com/AI-Hypercomputer/xpk.git
+
+python3 xpk/xpk.py cluster create \
+--cluster ${CLUSTER_NAME} \
+--cluster-cpu-machine-type=${MACHINE_TYPE} \
+--num-slices=${NUM_SLICES} \
+--tpu-type=${TPU_TYPE} \
+--enable-mtc \
+--enable-gcsfuse-csi-driver \
+--mtc-ramdisk-size=${RAMDISK_SIZE} \
+--mtc-gcs-bucket=${OUTPUT_PATH} \
+--gke-version=${GKE_VERSION}
+
+```
+
+---
+## Maxtext Configuration
+
+This configuration manages a `two-tiered checkpointing` system designed for both durability and rapid recovery.
```

Comment:
```suggestion
MaxText provides a set of configuration flags to control checkpointing options. This configuration manages a `two-tiered checkpointing` system designed for both durability and rapid recovery.

These flags can be set in the [`MaxText/configs/base.yml` file](https://github.com/AI-Hypercomputer/maxtext/blob/main/MaxText/configs/base.yml) or through the command line.
```

Please confirm if this is correct; my suggestion is to add more context to the readers about how to use these flags.